src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -250,23 +252,23 @@ public void run() {
     driver.run();
   }
 
-  private boolean launchNode(SchedulerDriver driver, Offer offer,
-        String nodeName, List<String> taskTypes, String executorName) {
+  private void launchNode(SchedulerDriver driver, Offer offer,
+        String nodeName, Map<String,String> taskInfo, String executorName) {
     //nodeName is the type of executor to launch
     //executorName is to distinguish different types of nodes
-    //taskType is the type of task in mesos to launch on the node
-    //taskName is a name chosen to identify the task in mesos and mesos-dns (if used)
+    //taskInfo is taskType mapped to taskName
     log.info(String.format("Launching node of type %s with tasks %s", nodeName,
-        taskTypes.toString()));
+        taskInfo));
     String taskIdName = String.format("%s.%s.%d", nodeName, executorName,
         System.currentTimeMillis());
     List<Resource> resources = getExecutorResources();
     ExecutorInfo executorInfo = createExecutor(taskIdName, nodeName, executorName, resources);
     List<TaskInfo> tasks = new ArrayList<>();
-    for (String taskType : taskTypes) {
+    Set<Map.Entry<String,String>> taskItems = taskInfo.entrySet();
```

Comment:
@hi10-artist, this is a style issue. In most cases, good style is to try to avoid method chaining in java. When there is too much going on on one line it is challenging to debug and also effects readability. We are trying to keep this project consistent and clean and keep each line as simple as possible.